gateway/src/apicast/policy/routing/upstream_selector.lua

```diff
@@ -25,7 +25,13 @@ function _M.select(_, rules, context)
     local cond_is_true = rule.condition:evaluate(context)
 
     if cond_is_true then
-      return Upstream.new(rule.url)
+      local upstream = Upstream.new(rule.url)
+
+      if rule.host_header then
```

Comment:
I think we have to cover case of an empty value.

Reply:
Good point :+1:

Reply:
Done in https://github.com/3scale/apicast/pull/983/commits/50ddc3757d493d7fde93828afafbf58e5a9d2ba0